app/utils/date-formatting.js

```diff
@@ -24,8 +24,15 @@ function verifyTimeStamp(dateString) {
 }
 
 // Parses a string to a Moment
-function parseDateString(value) {
-    return value ? moment(verifyTimeStamp(value), parseDateFormats, true) : undefined;
+function parseDateString(value, offset) {
+    // We need the offset here, otherwise the date will be parsed
+    // in UTC timezone
+    moment.tz.setDefault(offset);
+
+    // If the user clears out the date input field, the value will be a moment by default. In that case,
+    // verifyTimeStamp() runs into an error. Therefore, we will just return the value in the parseDateFormats
+    // format.
+    return moment.isMoment(value) ? moment(value, parseDateFormats, true) : moment(verifyTimeStamp(value), parseDateFormats, true);
```

Comment:
I think in the post model the `publishedAtOffset` CP should return `null` when it's dependent property `publishedAt` is `null`. Currently it always returns a moment value no matter what it's dependent property value is.

If that change is made then the behaviour of `parseDateString` can be put back to how it was before:
```
return value ? moment(verifyTimeStamp(value), parseDateFormats, true) : undefined;
```